### PR TITLE
Remove unmanaged compiler warnings.

### PR DESCRIPTION
**Pull request by PhenomRetroShare**

### CODE REVIEW

File: retroshare.pri [hunk 1]
```diff
@@ -501,8 +501,16 @@ no_rs_deprecatedwarning {
 }
 
 no_rs_cppwarning {
+    QMAKE_CXXFLAGS += -Wno-unknown-warning-option
```
Comment (G10h4ck) [inline retroshare.pri:504]: Ok with this, while the others are concerning, I'd prefer we fix those other warnings and personally i take of them in the code i edit

**Review by G10h4ck — COMMENTED** [review]
See inline comment

File: retroshare.pri (hunk 1 above)
Comment (PhenomRetroShare) [reply retroshare.pri:504]: Me too but some not...
And now with QtCreator you have direct warning with clang while editing.
 This is only to show "important" warning not to lose in the crowd...
And we still can use "CONFIG+=rs_cppwarning".

File: retroshare.pri [hunk 2]
```diff
@@ -501,8 +501,16 @@ no_rs_deprecatedwarning {
...
     QMAKE_CXXFLAGS += -Wno-cpp
     QMAKE_CXXFLAGS += -Wno-inconsistent-missing-override
+    QMAKE_CXXFLAGS += -Wno-misleading-indentation
+    QMAKE_CXXFLAGS += -Wno-unused-parameter
```
Comment (csoler) [inline retroshare.pri:508]: this one is very useful. Sometimes helps you detect some bugs.

Comment (csoler) [reply retroshare.pri:508]: Actually all warnings below should stay. They *are* useful.

Comment (PhenomRetroShare) [reply retroshare.pri:508]: So why you don't merge PR fixing it? ;)
For now all of them pollute compilation output.

**Pull request closed without merging**